libs/core/src/main/java/org/elasticsearch/core/internal/provider/EmbeddedImplClassLoader.java

```diff
@@ -319,14 +428,15 @@ private static Map<JarMeta, CodeSource> getProviderPrefixes(ClassLoader parent,
             List<String> jars = reader.lines().toList();
             Map<JarMeta, CodeSource> map = new HashMap<>();
             for (String jar : jars) {
+                final CodeSource codeSource = codeSource(parent.getResource(providerPrefix + JAR_LISTING_FILE), jar);
                 final String jarPrefix = providerPrefix + "/" + jar;
-                JarMeta jam;
-                if (isMultiRelease(parent, jarPrefix)) {
-                    jam = new JarMeta(jarPrefix, true);
-                } else {
-                    jam = new JarMeta(jarPrefix, false);
+                URI rootURI = rootURI(codeSource.getLocation());
+                Path p = embeddedJarPath(Set.of(jarPrefix), rootURI)[0];
+                var scan = scanPackages(p);
+                if (p.getFileSystem().provider().getScheme().equals("jar")) {
+                    p.getFileSystem().close();
```

Comment:
do we need to close this in the case of an exception, ie in tests?

Reply:
Done.